jukebox/src/main/webapp/js/script.js

```diff
@@ -22,78 +21,128 @@ var firebaseConfig = {
-            } else {
-                document.getElementById('group_name').textContent = roomSnapshot.data().name + " (General)";
-            }
+            document.getElementById("group_name").textContent = roomSnapshot.data().name
         } else {
             window.location.href = "/pin_error.html"
         }
     })
-    document.getElementById("overlay_pin").textContent = queryParameters.get("pin")
 }
 
-//Displays pin on create_pin.html
-function presentPIN() {
-
-    const queryParameters = new URLSearchParams(window.location.search);
-    document.getElementById('PIN').textContent = queryParameters.get('pin');
-}
-
-//Redirects user to their room after entering/creating a PIN
-function joinGroup(isHost) {
-
-    const queryParameters = new URLSearchParams(window.location.search);
-    if (isHost) {
-        window.location.href = '/room_host.html?pin=' + queryParameters.get('pin');;
+//This function navigates the user based on their host status
+//It uses a promise and cannot store values within the promise externally
+function navigateToRoom(isCreatingRoom) {
+    var storedUid = localStorage['uid']
+    var queryPin
+    if (isCreatingRoom) {
+        const queryParameters = new URLSearchParams(window.location.search);
+        queryPin = queryParameters.get('pin')
     } else {
-        window.location.href = '/room_general.html?pin=' + document.getElementById('pin').value;
+        queryPin = document.getElementById('pin').value
     }
-}
-
-//Takes in a PIN and returns true if a room with the PIN already exists
-function pinTaken(pinCode) {
-
-    var roomRef = db.collection('rooms').doc(pinCode);
-    var roomExists = false;
-    roomRef.get().then(function(roomSnapshot) {
-            if (roomSnapshot.exists) {
-                roomExists = true;
+    if(!queryPin || !storedUid) {
+        window.location.href = '/general_error.html'
+    }
+    const userRef = db.collection('rooms').doc(queryPin).collection('users').doc(storedUid)
+    if(!userRef) {
+        window.location.href = '/general_error.html'
+    }
+    userRef.get().then(function(userSnapshot) {
+        //If the user does not yet exist in the db, then add and redirect them
+        if (!userSnapshot.exists) {
+            db.collection("rooms").doc(queryPin).collection("users").doc(storedUid).set( {
+                isHost: false
+            })
+            window.location.href = '/room_general.html?pin=' + queryPin;
+        } else {
+            //Redirects to host or general page
+            if(userSnapshot.data().isHost) {
+                window.location.href = '/room_host.html?pin=' + queryPin;
             } else {
-                roomExists = false;
+                window.location.href = '/room_general.html?pin=' + queryPin;
+            }
         }
     })
-    return roomExists;
+}
+
+//Displays pin on create_pin.html
+function presentPIN() {
+    const queryParameters = new URLSearchParams(window.location.search)
```

Comment:
I noticed that in some places you end your lines in semi-colons, but in others you don't. It isn't a bug or anything and I don't really mind, but it might help with clarity if you stick with one.

Reply:
Other than that I think everything looks fine!